sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/MessageBody.cs

```diff
@@ -143,23 +159,44 @@ public override IEnumerator<ReadOnlyMemory<byte>> GetEnumerator()
                 return _segments.GetEnumerator();
             }
 
-            private void Append(Data segment)
+            private void Append(IEnumerable<Data> dataSegments)
             {
+                int length = 0;
+                int numberOfSegments = 0;
+                List<ReadOnlyMemory<byte>> segments = null;
+                foreach (var segment in dataSegments)
+                {
+                    segments ??= dataSegments is IReadOnlyCollection<Data> readOnlyList
+                        ? new List<ReadOnlyMemory<byte>>(readOnlyList.Count)
+                        : new List<ReadOnlyMemory<byte>>();
+                    ReadOnlyMemory<byte> dataToAppend = segment.Value switch
+                    {
+                        byte[] byteArray => byteArray,
```

Comment:
Is this only necessary for the EagerCopying because this will already be done by the time CopyingOnConversion is used?

Reply:
Yes these are the data segment value cases we have previously optimized for when interacting with the amqp lib for the cases we know what we add to the value property (or at least that is how I remember things)